packages/dm-core-plugins/src/form/fields/BinaryField.tsx

```diff
@@ -0,0 +1,79 @@
+import React from 'react'
+import { useFormContext } from 'react-hook-form'
+import { TStringFieldProps } from '../types'
+import { Button, Label } from '@equinor/eds-core-react'
+import { useRegistryContext } from '../RegistryContext'
+import { ErrorResponse, useDMSS } from '@development-framework/dm-core'
+import { AxiosError } from 'axios'
+
+const getTarget = (initialValue: any) => {
+  const { idReference } = useRegistryContext()
+  if (initialValue['address'].includes('blob')) {
+    const address = initialValue['address'].replace('blob://', '')
+    return address.split('/', 2)
+  } else {
+    return [idReference?.split('/', 2)[0], initialValue['address']]
+  }
+}
+
+const DownloadBinary = (props: any) => {
+  const { fileId, namePath, displayLabel, initialValue } = props
+  const dmssAPI = useDMSS()
+  const [data_source_id, blob_id] = getTarget(initialValue)
+
+  const handleDownload = () => {
+    dmssAPI
+      .blobGetById({
+        dataSourceId: data_source_id,
+        blobId: blob_id,
+      })
+      .then((response: any) => {
+        const data = response.data
+        const blob = new Blob([data], { type: 'application/pdf' })
+        const fileURL = window.URL.createObjectURL(blob)
+        // Setting various property values
+        const alink = document.createElement('a')
+        alink.href = fileURL
+        alink.download = 'SamplePDF.pdf'
+        alink.click()
+      })
+      .catch((error: AxiosError<ErrorResponse>) => {
+        console.error(error)
+      })
+  }
+
+  return (
+    <div>
+      <Label label={displayLabel} />
+      <Button
+        namePath={namePath}
+        label={displayLabel}
+        // eslint-disable-next-line @typescript-eslint/no-empty-function
+        onChange={() => {}}
+        onClick={() => handleDownload()}
+      >
```

Comment:
```suggestion
      <Button
        namePath={namePath}
        label={displayLabel}
        onClick={() => handleDownload()}
      >
```

cannot see why onChange is needed

Reply:
fixed